### PR TITLE
Handle non blittable LOGFONT structs

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Adding support for blittable structs broke non-blittable structs passed to `Font.FromLogFont`. Try to blit and fall through to the marshaller for non blittable types.

Fixes: https://github.com/dotnet/winforms/issues/2136

We should consider for 3.1 as this has been reported by a customer.

### CODE REVIEW

File: src/System.Drawing.Common/src/System/Drawing/Font.Windows.cs [hunk 1]
```diff
@@ -311,17 +311,21 @@ public static unsafe Font FromLogFont(object lf, IntPtr hdc)
...
                 GCHandle handle = GCHandle.Alloc(lf, GCHandleType.Pinned);
                 Buffer.MemoryCopy((byte*)handle.AddrOfPinnedObject(), &logFont, nativeSize, nativeSize);
                 handle.Free();
             }
+            catch (ArgumentException)
+            {
+                // If the type wasn't blittable it won't be able to be pinned with GCHandle. Try
+                // to use the marshaller to copy the "native" representation instead.
+                //
+                // This happens for classes or structs that have objects non fixed arrays in them,
```
Comment (stephentoub) [inline src/System.Drawing.Common/src/System/Drawing/Font.Windows.cs]: "that have objects non fixed arrays"?  I'm having trouble parsing that... is it missing a word?

Comment (JeremyKuhne) [reply src/System.Drawing.Common/src/System/Drawing/Font.Windows.cs]: Apparently I don't write well when I'm tired. 😂

**Comment by safern** [comment]
cc: @danmosemsft this should be considered for 3.1

**Comment by danmosemsft** [comment]
@JeremyKuhne if you would like to backport please create a PR with a brief template for me.

**Comment by danmosemsft** [comment]
Note that 4pm tomorrow (Wed) is the cutoff for preview 2 -- after that it would need M3 approval..

**Comment by danmosemsft** [comment]
@merriemcgaw do you wish this be in 3.1?

**Comment by danmosemsft** [comment]
@stephentoub did you have any other feedback on this (just asking since there's a port request up)

**Pull request merged**